File: typescript/base-hmi-types/src/screens/controls/HmiSystemDiagnosisControl.ts
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiColor } from "../base/HmiColor.js";
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiSystemDiagnosisViewKind } from "./HmiSystemDiagnosisViewKind.js";

export class HmiSystemDiagnosisControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiSystemDiagnosisControl;
  }

  viewKind = HmiSystemDiagnosisViewKind.Unknown;
  wrapAround?: HmiProperty<boolean>;
  selectionBackgroundColor?: HmiProperty<HmiColor>;
  selectionForegroundColor?: HmiProperty<HmiColor>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiSystemDiagnosisControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public class HmiSystemDiagnosisControl : HmiControlWindowBase
{
    public HmiSystemDiagnosisControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiSystemDiagnosisControl;
    }

    public HmiSystemDiagnosisViewKind ViewKind { get; set; }

    public HmiProperty<bool>? WrapAround { get; set; }

    public HmiProperty<HmiColor>? SelectionBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? SelectionForegroundColor { get; set; }
}
